cmd/kubeadm/app/util/staticpod/utils.go

```diff
@@ -228,6 +228,11 @@ func ReadinessProbe(host, path string, port int, scheme v1.URIScheme) *v1.Probe
 	return createHTTPProbe(host, path, port, scheme, 0, 15, 3, 1)
 }
 
+// StartupProbe creates a Probe object with a HTTPGet handler
+func StartupProbe(host, path string, port int, scheme v1.URIScheme) *v1.Probe {
```

Comment:
With the introduction of a startup Probe we should remove the `InitialDelaySeconds` for the readiness and liveness probe since the startup probe ensures that the application has started up. Also I think we can drop the `InitialDelaySeconds` for the startup probe too.

Reply:
for both we can set them to InitialDelaySeconds = PeriodSeconds, to skip one period before running a check?

the readyness probe is what is responsible for setting the container status to "ready". for that we have 0, which is important because we don't want to delay user infrastructure checks looking for "ready" status on kubeadm static pods.